src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -161,6 +161,15 @@ class Plugin(RepositoryPlugin):
     def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
         return cls(sources=config.tdr_sources(catalog))
 
+    def __attrs_post_init__(self):
+        for source in self.sources:
+            source = TDRSource.parse(source)
+            api_project = self.tdr.project_for_source(source)
```

Comment:
I don't think we want to do this for every source every time the plugin is instantiated. Most clients (code that uses the plugin) cache their instances, but there may be many clients. Think about possible solutions and discuss them with me after stand-up and prior to coding one.